### ISSUE TITLE
Failed to install scikit-learn from ..?

### ISSUE THREAD

**Issue by Meesh011** [chaquo/chaquopy#1115]
Failed to install scikit-learn from https://files.pythonhosted.org/packages/da/b4/5d7b9f64d2c3795ac37da3d8bc303be7950e575ec21ac7d0eb9de34fe0d5/scikit-learn-1.4.1.post1.tar.gz#sha256=93d3d496ff1965470f9977d05e5ec3376fb1e63b10e4fda5e39d23c2d8969a30.

how do I overcome this problem..?

**Comment by mhsmith** [comment]
Our newest scikit-learn version is 1.3.2, but it's only available for Python 3.9 and 3.10. 

If anyone needs a newer version, please click the thumbs up button above, and post a comment explaining why you need it.

**Comment by Meesh011** [comment]
i have a model in .json that needs to be integrated through chaqoupy.. i need scikit and scipy compatible versions with python 12

**Comment by mhsmith** [comment]
Unfortunately librosa requires Numba, which we currently only support on Python 3.8 (https://github.com/chaquo/chaquopy/issues/834#issuecomment-1485745722).

**Comment by rarcher** [comment]
Would like to upgrade to a newer version of Python for security reasons.  Currently prevented from doing this by scikit_learn version limitations.

Edit:  Also, it seems there's a known security vulnerability present in the scikit_learn versions currently available for Chaquopy:  https://osv.dev/vulnerability/PYSEC-2024-110

**Comment by mhsmith** [comment]
I agree that the `scikit-learn` issue could be a security problem in some situations.

As for the Python version, you can see from [the repository](https://chaquo.com/pypi-13.1/) that we support `scipy` and `scikit-learn` up as far as Python 3.10. This Python version is still receiving security updates, and we'll be including its current release in the next version of Chaquopy (#1352).

**Comment by mhsmith** [comment]
I'm now working on adding official Android support to this package (https://github.com/scikit-learn/scikit-learn/pull/33055).

**Comment by capturetheventure** [comment]
I also just stumbled upon this security issue with the available scikit_learn version. I built [an app to run Jupyter Notebooks on Android](https://play.google.com/store/apps/details?id=app.callisto), all based on Chaquopy. scikit_learn, scipy and statsmodels are currently the only blockers to upgrade from Python 3.10 to 3.13. This would also fix the upcoming issues with the 16KB page size requirement.

**Comment by mhsmith** [comment]
Unfortunately it looks like this package won't be officially released for Android until next year. Meanwhile, I've uploaded some unofficial wheels for Python 3.13, which you can use by adding this line to the `pip` section of your build.gradle file:
```gradle
options("--extra-index-url", "https://chaquo.com/pypi-upstream/")
```
